drizzlepac/processInput.py

```diff
@@ -131,6 +131,23 @@ def setCommonInput(configObj, createOutwcs=True, overwrite_dict={}):
         # Update configObj with values from mpars
         cfgpars.mergeConfigObj(configObj, mdriztab_dict)
     
+    # add warnings for kernels that may not work as intended
+    kernels_w_warnings = ["gaussian", "lanczos3", "tophat"]
+    for kernel_param in overwrite_dict:
+        if overwrite_dict[kernel_param] in kernels_w_warnings:
+            if overwrite_dict[kernel_param] == "tophat":
+                log.warning(
+                    f'WARNING: It was found that kernel "{overwrite_dict[kernel_param]}" does not work as intended. '
+                    + f'It has been removed as a option. Using kernel "square" instead for {kernel_param}.'
```

Comment:
IMO, it is better to let the code raise an exception instead of changing the kernel: some may not read through the log file but an exception is noticeable

Reply:
removed.